exchanges/stream/websocket.go

```diff
@@ -1241,3 +1263,38 @@ func signalReceived(ch chan struct{}) bool {
 		return false
 	}
 }
+
+// GetConnection returns a connection by connection signature (defined in wrapper setup) for request and response
+// handling in a multi connection context.
+func (w *Websocket) GetConnection(connSignature any) (Connection, error) {
+	if w == nil {
+		return nil, fmt.Errorf("%w: %T", common.ErrNilPointer, w)
+	}
+
+	if connSignature == nil {
+		return nil, errConnectionSignatureNotSet
+	}
+
+	w.m.Lock()
+	defer w.m.Unlock()
+
+	if !w.IsConnected() {
+		return nil, ErrNotConnected
+	}
+
+	if !w.useMultiConnectionManagement {
+		return nil, fmt.Errorf("%s: multi connection management not enabled %w please use exported Conn and AuthConn fields", w.exchangeName, errCannotObtainOutboundConnection)
+	}
```

Comment:
I note the need for the mutex, but put this above IsConnected. None of the other checks matter if the feature isn't enabled